test/dynamo/test_model_output.py

```diff
@@ -137,6 +137,11 @@ class MyDataClass(ModelOutput):
             d: torch.Tensor = None
             e: torch.Tensor = None
 
+        # Patch the __module__; to avoid expensive imports we currently only
+        # trigger custom logic for ModelOutput and BaseModel when the module
+        # names match.
+        MyDataClass.__module__ = "transformers.modeling_outputs"
+
```

Comment:
I'm curious if we wrap this dataclass object as ```CustomDictVariable``` and test it.

Reply:
We currently wrap it as a `DataClassVariable` - it's actually impossible to wrap any value as a `CustomDictVariable` at this point, which is one of the cluster of small but related bugs I'm looking at.

I've extended the test locally to actually construct a value as well as passing one in, in which case you'll wind up in a state where the passed-in value is a `DataClassVariable` and the constructed one is a `CustomDictVariable`; the test actually fails if we do that because the `CustomDictVariable` doesn't model this behavior correctly.

I'm not sure how to make it evident from the test which one we are getting. Unless you force the compiler to behave differently you can't actually see the inconsistency between `wrap` and `create`